source/dem/particle_wall_linear_force.cc

```diff
@@ -214,6 +215,187 @@ ParticleWallLinearForce<dim>::calculate_particle_wall_contact_force(
+                std::get<2>(particle_triangle_information);
+
+              unsigned int particle_counter = 0;
+
+              for (auto &&contact_info : map_info | boost::adaptors::map_values)
+                {
+                  // If particle passes the distance check
+                  if (pass_distance_check[particle_counter])
+                    {
+                      // Define the total force of contact, properties of
+                      // particle as local parameters
+                      auto &particle            = contact_info.particle;
+                      auto &particle_properties = particle->get_properties();
+
+                      const Point<3> &projection_point =
+                        projection_points[particle_counter];
+
+                      Point<3> particle_location_3d;
+
+                      if constexpr (dim == 3)
+                        particle_location_3d = particle->get_location();
+
+                      if constexpr (dim == 2)
+                        particle_location_3d =
+                          point_nd_to_3d(particle->get_location());
+
+                      const double particle_triangle_distance =
+                        particle_location_3d.distance(projection_point);
+
+                      // Find normal overlap
+                      double normal_overlap =
+                        ((particle_properties[DEM::PropertiesIndex::dp]) *
+                         0.5) -
+                        particle_triangle_distance;
+
+                      if (normal_overlap > 0)
+                        {
+                          contact_info.normal_overlap = normal_overlap;
+                          // Update normal vector
+                          contact_info.normal_vector =
+                            normal_vectors[particle_counter];
+                          contact_info.point_on_boundary = projection_point;
+
+                          contact_info.boundary_id = solid_counter;
+
+                          this
+                            ->update_particle_floating_wall_contact_information(
+                              contact_info,
+                              particle_properties,
+                              dt,
+                              translational_velocity,
+                              rotational_velocity,
+                              center_of_rotation.distance(
+                                particle_location_3d));
+
+                          // This tuple (forces and torques) contains four
+                          // elements which are: 1, normal force, 2, tangential
+                          // force, 3, tangential torque and 4, rolling
+                          // resistance torque, respectively
+                          std::tuple<Tensor<1, 3>,
+                                     Tensor<1, 3>,
+                                     Tensor<1, 3>,
+                                     Tensor<1, 3>>
```

Comment:
That's a complicated tuple... I would make a typedef. Like forces_torques_containers and it would be a type def that points to that kind of tuple.

Reply:
I use the same tuple for particle-wall and particle-floating wall contacts. I have to change them all. Can I do it in another PR?

Reply:
Yup